source/MaterialXGenGlsl/GlslResourceBindingContext.cpp

```diff
@@ -26,30 +32,52 @@ void GlslResourceBindingContext::emitDirectives(GenContext& context, ShaderStage
     }
 }
 
-void GlslResourceBindingContext::emitUniformBlock(GenContext& context, const VariableBlock& uniforms, SyntaxPtr syntax, ShaderStage& stage)
+void GlslResourceBindingContext::emitResourceBindings(GenContext& context, const VariableBlock& uniforms, ShaderStage& stage)
 {
     ShaderGenerator& generator = context.getShaderGenerator();
+    const Syntax& syntax = generator.getSyntax();
 
-    generator.emitString("layout (binding=" + std::to_string(_hwBindLocation) + ") " + syntax->getUniformQualifier() + " " + uniforms.getName(), stage);
-    generator.emitScopeBegin(stage);
-    generator.emitVariableDeclarations(uniforms, EMPTY_STRING, Syntax::SEMICOLON, context, stage, false);
-    generator.emitScopeEnd(stage, true);
-    generator.emitLineBreak(stage);
-    _hwBindLocation++;
-}
-
-void GlslResourceBindingContext::emitSamplerBlocks(GenContext& context, const VariableBlock& uniforms, SyntaxPtr syntax, ShaderStage& stage)
-{
-    ShaderGenerator& generator = context.getShaderGenerator();
+    // First, emit all value uniforms in a block with single layout binding
```

Comment:
Instead of keeping a separate variable block for sampler uniforms the split into value vs sampler uniforms is done here when emitting the code.